build_tools/bazel_to_cmake/bazel_to_cmake.py

```diff
@@ -86,15 +95,40 @@ def setup_environment():
       os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 
 
+def repo_relpath(path):
+  return os.path.relpath(path, repo_root)
+
+
 def log(*args, **kwargs):
   print(*args, **kwargs, file=sys.stderr)
 
 
 def convert_directory_tree(root_directory_path, write_files,
                            allow_partial_conversion):
-  log(f"convert_directory_tree: {root_directory_path}")
-  for root, _, _ in os.walk(root_directory_path):
-    convert_directory(root, write_files, allow_partial_conversion)
+  log(f"Converting directory tree rooted at: {root_directory_path}")
+  convert_directories((root for root, _, _ in os.walk(root_directory_path)),
+                      write_files, allow_partial_conversion)
+
+
+def convert_directories(generator, write_files, allow_partial_conversion):
```

Comment:
Could you add a note on why you chose to split this into two functions? It seems like you could move the contents of `convert_directory_tree` (logging and creating this generator) after line 200, and then rename `generator` in `convert_directories` to `directories` as not all calls `convert_directories` pass a generator.

Reply:
Yeah convert_directory_tree could probably be inlined at this point. It's a separate function because it was already a separate function. I created the separate convert_directories because I wanted to share the logic around error handling. I think I called it `generator` to emphasize that it just needed to be iterable, thinking that a list was also a kind of degenerate generator, but that terminology might totally mismatch with python conventions.